dist/hijax.js

```diff
@@ -69,25 +69,22 @@
 
     // Get a particular response header value by key
     // Extracted from jQuery 2.1.1
+    // Changed to grab content-type only (we only use content-type, no need to set everything) from XHR object
     Hijacker.prototype.getResponseHeader = function(xhr, key) {
         var rHeaders = /^(.*?):[ \t]*([^\r\n]*)$/mg;
-        var match;
         var responseHeadersString;
+        var contentType;
 
         if ([states.HEADERS_RECEIVED, states.LOADING, states.DONE]
                 .indexOf(xhr.readyState) > -1) {
 
             responseHeadersString = xhr.getAllResponseHeaders();
 
-            if (!this.responseHeaders) {
-                this.responseHeaders = {};
-                while ((match = rHeaders.exec(responseHeadersString))) {
-                    this.responseHeaders[match[1].toLowerCase()] = match[2];
-                }
-            }
-            match = this.responseHeaders[key.toLowerCase()];
+            contentType = responseHeadersString.match(/Content-Type:\s*(.*?);\s*charset/);
```

Comment:
@henryh15 this regexp will fail if the header name is mixed or lower case eg. `content-type`. It will also fail if the header doesn't specify a charset. eg. `Content-Type: text/html` is valid and this won't match.